calculator_test.go

```diff
@@ -2,23 +2,170 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	var testCases = []struct {
+		name string
+		a, b float64
+		nums []float64
+		want float64
+	}{
+		{name: "positive", a: 2, b: 2, want: 4},
+		{name: "negative", a: -2, b: 5, want: 3},
+		{name: "one zero", a: 0, b: 3, want: 3},
+		{name: "two zeroes", a: 0, b: 0, want: 0},
+		{name: "decimal", a: 2.5, b: 4, want: 6.5},
+		{name: "many numbers", a: 1, b: 2, nums: []float64{3, 4, 5, 6, 7, 8, 9, 10}, want: 55},
+	}
+	for _, tc := range testCases {
+		got := calculator.Add(tc.a, tc.b, tc.nums...)
+		if tc.want != got {
+			t.Errorf("test name: %s, want %f, got %f", tc.name, tc.want, got)
+		}
+	}
+}
+
+func TestAddRandom(t *testing.T) {
+	t.Parallel()
+	rand.Seed(time.Now().UnixNano())
+
+	for i := 0; i < 1000; i++ {
+		a, b := rand.Float64(), rand.Float64()
+		want := a + b
+		got := calculator.Add(a, b)
+		if want != got {
+			t.Errorf("adding %f, %f: want %f, got %f", a, b, want, got)
+		}
+	}
+}
+
+func TestSubtract(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		name string
+		a, b float64
+		nums []float64
+		want float64
+	}{
+		{name: "positive", a: 2, b: 2, want: 0},
+		{name: "negative", a: -2, b: 5, want: -7},
+		{name: "one zero", a: 0, b: 3, want: -3},
+		{name: "two zeroes", a: 0, b: 0, want: 0},
+		{name: "decimal", a: 2.5, b: 4, want: -1.5},
+		{name: "many numbers", a: 1, b: 2, nums: []float64{3, 4, 5, 6, 7, 8, 9, 10}, want: -53},
+	}
+	for _, tc := range testCases {
+		got := calculator.Subtract(tc.a, tc.b, tc.nums...)
+		if tc.want != got {
+			t.Errorf("test name: %s, want %f, got %f", tc.name, tc.want, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		name string
+		a, b float64
+		nums []float64
+		want float64
+	}{
+		{name: "positive", a: 2, b: 2, want: 4},
+		{name: "negative", a: -2, b: 5, want: -10},
+		{name: "one zero", a: 0, b: 3, want: 0},
+		{name: "two zeroes", a: 0, b: 0, want: 0},
+		{name: "decimal", a: 2.5, b: 4, want: 10},
+		{name: "many numbers", a: 1, b: 2, nums: []float64{3, 4, 5, 6, 7, 8, 9, 10}, want: 3628800},
+		{name: "many of the same numbers", a: 2, b: 2, nums: []float64{2, 2, 2, 2, 2, 2, 2, 2}, want: 1024},
+	}
+	for _, tc := range testCases {
+		got := calculator.Multiply(tc.a, tc.b, tc.nums...)
+		if tc.want != got {
+			t.Errorf("test name: %s, want %f, got %f", tc.name, tc.want, got)
+		}
+	}
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		name        string
+		a, b        float64
+		nums        []float64
+		want        float64
+		errExpected bool
+	}{
+		{name: "positive", a: 2, b: 2, want: 1, errExpected: false},
```

Comment:
Try a test case like `2/3`, because that's a number that can't be accurately represented in floating-point.

Reply:
Addressed in https://github.com/micuffaro/ftl-fundamentals/pull/1/commits/48b6d6bff1d8adc4e283472874dfa3f90d002132